src/HTML5/Parser/Tokenizer.php

```diff
@@ -313,12 +313,13 @@ protected function eof($tok)
      */
     protected function characterReference()
     {
-        $ref = $this->decodeCharacterReference();
-        if ($ref !== false) {
-            $this->buffer($ref);
-            return true;
+        if ($this->scanner->current() !== '&') {
+            return false;
         }
-        return false;
+
+        $ref = $this->decodeCharacterReference();
```

Comment:
I don't think this variable is necessary, is it?

Reply:
it used on the next line by the buffer function

Reply:
I think @tgalopin meant you could just use `$this->buffer(this->decodeCharacterReference());` instead.

Reply:
aaa, not a big deal